lib/lapin/connection.ex

```diff
@@ -612,4 +622,18 @@ defmodule Lapin.Connection do
       {:error, :missing_params, Enum.reject(params, &Keyword.has_key?(configuration, &1))}
     end
   end
+
+  defp monitor_processes(connection, producers, consumers) do
+    Process.monitor(connection.pid)
+
+    Enum.each(producers, fn producer ->
+      Process.monitor(producer.channel.pid)
+    end)
+
+    Enum.each(consumers, fn consumer ->
+      Process.monitor(consumer.channel.pid)
+    end)
```

Comment:
There's a single `:DOWN` handler in this module and it stops and reconnects. The entire connection. This means any consumer or producer dying will take down all the other ones 😄 IMO we should check which process died and only restart this one in case it's one of the channels.